File: 005_longest_palindrom_substring/solution.go
package main

import (
	"fmt"
)

func main() {
	res := longestPalindrome("babbbbad")
	fmt.Println(res)

}

func longestPalindrome(s string) string {
	max_length := ""
	for i := 0; i < len(s); i++ {
		j := i
		k := i
		for j >= 0 && k < len(s) && s[j] == s[k] {
			word := s[j : k+1]
			k++
			j--
			if len(word) > len(max_length) {
				max_length = word
			}
		}

		j = i
		k = i + 1
		for j >= 0 && k < len(s) && s[j] == s[k] {
			word := s[j : k+1]
			k++
			j--
			if len(word) > len(max_length) {
				max_length = word
			}
		}

	}
	return max_length

}

File: 005_longest_palindrom_substring/solution.py
class Solution(object):
    def longestPalindrome(self, s):
        word = ""
        for i in range(len(s)):
            current_word = ""
            j =  k = i
            while (j >= 0 and k < len(s) and s[j] == s[k]): 
                current_word = s[j:k+1]
                j -= 1
                k += 1
            if len(current_word) > len(word):
                word = current_word
                
            
            current_word = ""
            j = i
            k = i + 1
            while (j >= 0 and k < len(s) and s[j] == s[k]): 
                current_word = s[j:k+1]
                j -= 1
                k += 1
            if len(current_word) > len(word):
                word = current_word
                

            
        return word
    
    
    
    
a = Solution()

print(a.longestPalindrome("cbbbbd"))


